src/app/campaign.model.ts

```diff
@@ -42,13 +46,17 @@ export class Campaign {
     public target: number,
     public title: string,
     public updates: Array<{content: string, modifiedDate: Date}>,
+    public usesSharedFunds: boolean,
     public alternativeFundUse?: string,
     public campaignCount?: number,
     public championOptInStatement?: string,
     public championRef?: string,
     public feePercentage?: number,
     public logoUri?: string,
+    public parentAmountRaised?: number,
+    public parentDonationCount?: number,
     public parentRef?: string,
+    public parentTarget?: number,
```

Comment:
are these three new fields always non-null if and only if parentUsesSharedFunds is true? In which case we could make that clear in the type system by putting them together in optional subj-object?

Something like:

```diff
     public isMatched: boolean,
     public matchFundsRemaining: number,
     public matchFundsTotal: number,
-    public parentUsesSharedFunds: boolean,
     public problem: string,
     public quotes: Array<{person: string, quote: string}>,
     public ready: boolean,
@@ -53,6 +52,11 @@ export class Campaign {
     public championRef?: string,
     public feePercentage?: number,
     public logoUri?: string,
+    public sharedFundsParent?: {
+        amountRaised: number,
+        donationCount: number,
+        target: number,
+    },
     public parentAmountRaised?: number,
     public parentDonationCount?: number,
     public parentRef?: string,
```

Reply:
@bdsl I think they are supplied whenever applicable, since that's both faster on the SF side and lets us use the data in other ways later if we want. (e.g. a donate page could show both totals)

So we could do this and call it `parentCampaign` possibly, but as I mentioned elsewhere it feels like it's a limited benefit for this many properties.

Reply:
ah ok. I was meaning to go and have another look at that SF PR. It's less useful if we still need the boolean.